### PR TITLE
[Batch Mode] Pass PrimarySpecificPaths to createEmptyModule and IRGenModule()

### PR DESCRIPTION
**Pull request by davidungar**
Must be merged along with https://github.com/apple/swift/pull/14230 after both PRs have been reviewed and polished.

### CODE REVIEW

File: source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -19,25 +19,26 @@
 #include <sstream>
 
 #include "swift/AST/ASTContext.h"
+#include "swift/AST/ASTMangler.h"
 #include "swift/AST/DebuggerClient.h"
 #include "swift/AST/Decl.h"
 #include "swift/AST/DiagnosticEngine.h"
 #include "swift/AST/ExistentialLayout.h"
-#include "swift/AST/IRGenOptions.h"
-#include "swift/AST/ASTMangler.h"
```
Comment (dcci) [inline source/Symbol/SwiftASTContext.cpp]: Please submit the header sorting separately.

File: source/Symbol/SwiftASTContext.cpp [hunk 2]
```diff
@@ -4556,8 +4557,8 @@ swift::SILModule *SwiftASTContext::GetSILModule() {
   VALID_OR_RETURN(nullptr);
 
   if (m_sil_module_ap.get() == NULL)
-    m_sil_module_ap = swift::SILModule::createEmptyModule(GetScratchModule(),
-                                                          GetSILOptions());
+    m_sil_module_ap = swift::SILModule::createEmptyModule(
+                                                          GetScratchModule(), GetSILOptions(), swift::PrimarySpecificPaths("", "<lldb>"));
```
Comment (dcci) [inline source/Symbol/SwiftASTContext.cpp]: Can you please `clang-format` your changes?

**Review by dcci — COMMENTED** [review]
Sorry, I must have missed this one. Anyway, I see you didn't merge the compiler bits yet, so here's a quick review

File: source/Symbol/SwiftASTContext.cpp (hunk 1 above)
Comment (davidungar) [reply source/Symbol/SwiftASTContext.cpp]: Thanks for looking at this. I have no need to reorder the headers, it's just what git-clang-format did. So, I'll revise the PR to only add the new header.

File: source/Symbol/SwiftASTContext.cpp (hunk 2 above)
Comment (davidungar) [reply source/Symbol/SwiftASTContext.cpp]: Yup, sorry for the oversight.

**Comment by davidungar** [comment]
Have addressed the concerns and pushed.

**Comment by dcci** [comment]
Thanks David. The patch makes sense to me. 
I followed batch mode (but not particularly closely, I have to admit). I wonder if this can have impacts on debuggability? (I honestly don't expect so, as if this is the only change, it's just an API change), but I would like to confirm with you, so we can plan in advance changes, if needed.

**Comment by davidungar** [comment]
My pleasure, Davide. I appreciate the support in checking my work. 
Impacts on debuggability? I hope not, but would be happy to discuss it. The intention of these changes is to have no impact at all, but there's often a slip twixt the cup and the lip.

**Comment by dcci** [comment]
I really need to double check this (if we use for anything real) but you're swapping the assignment of a member variable to an assignment to a temporary (for `mainInputFileName`).
So, I'll take a closer look tomorrow, but, if we need this to be set, for, e.g., IRGen, then this change has a small bug (if all the tests still passes, well, we should add a test for it).

Otherwise, we might nuke that member variable entirely. You really don't need to do that, I can take care of it, but your change might help us discovering some unused code.

**Comment by davidungar** [comment]
The issue is that I'm trying to move some state out of IRGenOpts: MainInputFile and OutputFIlenames. CompilerInstance owns a CompilerInvocation which owns an IRGenOpts. So the status quo is that MainInputFIle and OutputFilenames are owned by CompilerInstance. But this does not work when there are multiple primaries, since each primary determines the MainInputFile, and logically, the OutputFilename. So, these entities migrate to parameters to the SILModule and IRGenModule. Since there are other such filenames as well, I'm introducing PrimarySpecificFilenames to hold all such.
The debugger changes are intended to preserve its behavior while letting me remove those instance variables from IRGenOpts.

**Comment by dcci** [comment]
I took a closer look at this after your comment and I guess this should be safe. The changes look good.

**Comment by davidungar** [comment]
Please test with the following PR:
https://github.com/apple/swift/pull/14230

@swift-ci Please test

**Comment by davidungar** [comment]
@dcci Davide, the compiler bits are ready to merge as soon as the tests pass. (I'm having trouble with the compiler side not getting the latest changes I've made to this PR.)
There have been some very small changes to this code since you looked at it, just passing two strings instead of a PrimarySpecificPaths structure. Do you want to look it over again? And/or approve the changes? Or do you want me to just merge them as soon as the tests pass, since you did write that the changes look good. Thanks.

**Comment by davidungar** [comment]
I've looked over the changes again, and they are quite simple. I'm going to land them so I can land the big changes in the frontend.

**Pull request merged**